cmd/bosun/conf/conf.go

```diff
@@ -324,1175 +260,122 @@ func GetNotificationChains(c *Conf, n map[string]*Notification) [][]string {
-		c.InfluxConfig.Timeout = d
-	case "httpListen":
-		c.HTTPListen = v
-	case "hostname":
-		c.Hostname = v
-	case "relayListen":
-		c.RelayListen = v
-	case "smtpHost":
-		c.SMTPHost = v
-	case "smtpUsername":
-		c.SMTPUsername = v
-	case "smtpPassword":
-		c.SMTPPassword = v
-	case "emailFrom":
-		c.EmailFrom = v
-	case "stateFile":
-		c.StateFile = v
-	case "ping":
-		c.Ping = true
-	case "pingDuration":
-		d, err := time.ParseDuration(v)
-		if err != nil {
-			c.errorf(err.Error())
-		}
-		c.PingDuration = d
-	case "noSleep":
-		c.NoSleep = true
-	case "unknownThreshold":
-		i, err := strconv.Atoi(v)
-		if err != nil {
-			c.error(err)
-		}
-		c.UnknownThreshold = i
-	case "timeAndDate":
-		sp := strings.Split(v, ",")
-		var t []int
-		for _, s := range sp {
-			i, err := strconv.Atoi(strings.TrimSpace(s))
-			if err != nil {
-				c.error(err)
-			}
-			t = append(t, i)
-		}
-		c.TimeAndDate = t
-	case "responseLimit":
-		i, err := strconv.ParseInt(v, 10, 64)
-		if err != nil {
-			c.error(err)
-		}
-		if i <= 0 {
-			c.errorf("responseLimit must be > 0")
-		}
-		c.ResponseLimit = i
-	case "defaultRunEvery":
-		var err error
-		c.DefaultRunEvery, err = strconv.Atoi(v)
-		if err != nil {
-			c.error(err)
-		}
-		if c.DefaultRunEvery <= 0 {
-			c.errorf("defaultRunEvery must be > 0")
-		}
-	case "searchSince":
-		s, err := opentsdb.ParseDuration(v)
-		if err != nil {
-			c.error(err)
-		}
-		c.SearchSince = s
-	case "unknownTemplate":
-		c.unknownTemplate = v
-		t, ok := c.Templates[c.unknownTemplate]
-		if !ok {
-			c.errorf("template not found: %s", c.unknownTemplate)
-		}
-		c.UnknownTemplate = t
-	case "squelch":
-		c.squelch = append(c.squelch, v)
-		if err := c.Squelch.Add(v); err != nil {
-			c.error(err)
-		}
-	case "shortURLKey":
-		c.ShortURLKey = v
-	case "internetProxy":
-		c.InternetProxy = v
-	case "ledisDir":
-		c.LedisDir = v
-	case "ledisBindAddr":
-		c.LedisBindAddr = v
-	case "redisHost":
-		c.RedisHost = v
-	case "redisPassword":
-		c.RedisPassword = v
-	case "redisDb":
-		i, err := strconv.Atoi(v)
-		if err != nil {
-			c.error(err)
-		}
-		c.RedisDb = i
-	case "annotateElasticHosts":
-		c.AnnotateElasticHosts = strings.Split(v, ",")
-	case "annotationIndex":
-		c.AnnotateIndex = v
-	case "minGroupSize":
-		i, err := strconv.Atoi(v)
-		if err != nil {
-			c.error(err)
-		}
-		c.MinGroupSize = i
-	default:
-		if !strings.HasPrefix(k, "$") {
-			c.errorf("unknown key %s", k)
-		}
-		c.Vars[k] = v
-		c.Vars[k[1:]] = c.Vars[k]
-	}
+type Entry struct {
+	*ExprEntry
+	Def  string
+	Name string
 }
 
-func (c *Conf) loadSection(s *parse.SectionNode) {
-	switch s.SectionType.Text {
-	case "template":
-		c.loadTemplate(s)
-	case "alert":
-		c.loadAlert(s)
-	case "notification":
-		c.loadNotification(s)
-	case "macro":
-		c.loadMacro(s)
-	case "lookup":
-		c.loadLookup(s)
-	default:
-		c.errorf("unknown section type: %s", s.SectionType.Text)
-	}
+type Macro struct {
+	Text     string
+	Pairs    interface{} // this is BAD TODO
```

Comment:
Before it was non exported, nodePair, `Pairs []nodePair`, so this was a shortcut to get it working when I moved the rule stuff to a different package